src/services/expenses.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+let expenses = [];
+
+const getAllExpenses = () => {
+  return expenses;
+};
+
+const getExpenseById = (expenseId) => {
+  const foundExpense = expenses.find(({ id }) => id === +expenseId);
+
+  return foundExpense || null;
+};
+
+const createExpense = (data) => {
+  const maxId = Math.max(...expenses.map(({ id }) => id), 0);
+  const newExpense = Object.assign({ id: maxId + 1 }, data);
+
+  expenses.push(newExpense);
+
+  return newExpense;
+};
+
+const removeExpense = (expenseId) => {
+  expenses = expenses.filter(({ id }) => id !== +expenseId);
+};
+
+const updateExpense = (userId, data) => {
+  const foundExpense = getExpenseById(+userId);
+
+  Object.assign(foundExpense, data);
```

Comment:
Unsafe. Better check if exists.

Reply:
But I check it in the controller or is it not enough? Wouldn't checking here violate the principle of fail-fast?

Reply:
If you have checked it before - it's ok.
For me it's unsafe because if foundExpense in some rare case will be `null` - your script will fail. Your controller and service - they are independent. You can switch to another freely. But if you forget to check it next time - it can cause a problem.